client/src/tests/MovieContainer.test.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { mount } from 'enzyme';
+import MovieContainer from '../components/MoviesContainer';
+import movies from '../components/movies';
+
+it ('renders correctly', ()=>{
+    const component = renderer.create(<MovieContainer />);
+
+    var jsonComponent = component.toJSON();
+    expect(jsonComponent).toMatchSnapshot();
+});
+
+it ('loads movie details when some movie is selected', ()=>{
+    const component = mount(<MovieContainer />);
+    const movieslist = component.find('.movie')
+    expect(component.find('.movie').length).toEqual(7);
+    expect(component.state('selectedMovie')).toBeNull();
+    movieslist.at(0).simulate('click');
```

Comment:
_minor_

and you can keep splitting this into smaller unit tests, which will have no more than 1-2 assertions and verify exactly that what is specified in a test case.

```javascript
it('should select the movie', () => {
  const component = mount(<MovieContainer />);
  const movieslist = component.find('.movie');
  movieslist.at(0).simulate('click');

  expect(component.state('selectedMovie')).toEqual(movies.data[6]);
});
```